src/data.cc

```diff
@@ -0,0 +1,140 @@
+/*
+ * data.cc - Game resources file functions
+ *
+ * Copyright (C) 2014  Wicked_Digger <wicked_digger@mail.ru>
+ *
+ * This file is part of freeserf.
+ *
+ * freeserf is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * freeserf is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with freeserf.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "src/data.h"
+
+#include <cstdlib>
+
+#include "src/misc.h"
+BEGIN_EXT_C
+  #include "src/log.h"
+END_EXT_C
+#include "src/data-source-dos.h"
+
+data_t *data_t::instance = NULL;
+
+data_t::data_t() {
+  data_source = NULL;
+  instance = this;
+}
+
+data_t::~data_t() {
+  if (data_source != NULL) {
+    delete data_source;
+    data_source = NULL;
+  }
+
+  instance = NULL;
+}
+
+data_t *
+data_t::get_instance() {
+  if (instance == NULL) {
+    instance = new data_t();
+  }
+
+  return instance;
+}
+
+void
+data_t::add_to_search_paths(const char *path,
+                            const char *suffix) {
+  if (path == NULL) {
+    return;
+  }
+
+  std::string res_path = path;
+  if (suffix != NULL) {
+    res_path += '/';
+    res_path += suffix;
+  }
+
+  search_paths.push_back(res_path);
+}
+
+#define MAX_DATA_PATH      1024
+
+/* Load data file from path is non-NULL, otherwise search in
+ various standard paths. */
```

Comment:
@jonls window have no debug version of runtime libs by default, I advised him to download release binary.